src/hotspot/share/ci/ciEnv.cpp

```diff
@@ -1141,17 +1141,15 @@ void ciEnv::register_method(ciMethod* target,
 #endif
 
       if (entry_bci == InvocationEntryBci) {
-        if (TieredCompilation) {
```

Comment:
Just a naive question. Why this check has been removed?

Reply:
We want to let replacement of C2 method version by another C2 version of the same method in both tired and non-tired mode, which was not allowed